packages/klighd-core/src/views.tsx

```diff
@@ -107,24 +168,14 @@ export class KNodeView implements IView {
             node.opacity = 0.1
             rendering = getRendering(node.data, node, new KStyles, ctx, this.mListener)
         }
-        node.shadow = isShadow
 
-        if (node.id === '$root') {
-            // The root node should not be rendered, only its children should.
-            const children = ctx.renderChildren(node)
-            // Add all color and shadow definitions put into the context by the child renderings.
-            const defs = <defs></defs>
-            ctx.renderingDefs.forEach((value: VNode) => {
-                (defs.children as (string | VNode)[]).push(value)
-            })
+        node.shadow = isShadow
 
-            result.push(defs)
-            if (interactiveNodes) {
-                result.push(interactiveNodes)
-            }
-            result.push(...children)
+        let shadow = undefined
 
-            return <g>{...result}</g>
+        if (isShadow) {
+            // Render shadow of the node
+            shadow = getRendering(node.data, node, new KStyles, ctx, this.mListener)
```

Comment:
what are these changes for? Looks like some lines switched positions here.

Reply:
yes, that is basically what is done here.
this is done so that we don't need to get the rendering for the root node which would just be discarded in that case

Reply:
this change is now reverted as apparently the root node depends on a side-effect of the getRendering call when rendering elkt diagrams